modules/auxiliary/scanner/http/title.rb

```diff
@@ -0,0 +1,102 @@
+        HTML title attribute and location header (if set). Useful for rapidly identifying
+        interesting web applications en mass.
+      },
+      'Author'       => 'Stuart Morgan <stuart.morgan[at]mwrinfosecurity.com>',
+      'License'     => MSF_LICENSE,
+    )
+
+    register_options(
+      [
+        OptBool.new('STORE_NOTES', [ true, 'Store the captured information in notes. Use "notes -t http.title" to view', true ]),
+        OptBool.new('SHOW_ERRORS', [ true, 'Show error messages relating to grabbing titles on the console', true ]),
+        OptBool.new('SHOW_TITLES', [ true, 'Show the titles on the console as they are grabbed', true ])
+      ], self.class)
+
+    deregister_options('VHOST')
+  end
+
+  def run
+    if datastore['STORE_NOTES'] == false && datastore['SHOW_ERRORS'] == false && datastore['SHOW_TITLES'] == false
+      print_error("Notes storage is false, errors have been turned off and titles are not being shown on the console. There isn't much point in running this module.")
+    else
+      super
+    end
+  end
+
+  def run_host(target_host)
+    begin
+        # Send a normal GET request
+        res = send_request_cgi(
+          'uri' => '/',
+          'method' => 'GET'
+        )
+
+        # If no response, quit now
+        if res.nil?
+          print_error("[#{target_host}:#{rport}] No response") if datastore['SHOW_ERRORS'] == true
+          return
+        end
+
+        # Retrieve the headers to capture the Location and Server header
+        # Note that they are case-insensitive but stored in a hash
+        server_header = nil
+        location_header = nil
+        if !res.headers.nil?
+          res.headers.each do |key, val|
+            location_header = val if key.downcase == 'location'
+            server_header  = val if key.downcase == 'server'
+          end
+        else
+          print_error("[#{target_host}:#{rport}] No HTTP headers") if datastore['SHOW_ERRORS'] == true
+        end
+
+        # If the body is blank, just stop now as there is no chance of a title
+        if res.body.nil?
+          print_error("[#{target_host}:#{rport}] No webpage body") if datastore['SHOW_ERRORS'] == true
+          return
+        end
+
+        # Very basic, just match the first title tag we come to. If the match fails,
+        # there is no chance that we will have a title
+        rx = %r{<title>[\n\t\s]*(?<title>.+?)[\s\n\t]*</title>}im.match(res.body.to_s)
+        unless rx
+          print_error("[#{target_host}:#{rport}] No webpage title") if datastore['SHOW_ERRORS'] == true
+          return
+        end
+
+        # Last bit of logic to capture the title
+        rx[:title].strip!
+        if rx[:title] != ''
+          rx_title = Rex::Text.html_decode(rx[:title])
+          print_status("[#{target_host}:#{rport}] [C:#{res.code}] [R:#{location_header}] [S:#{server_header}] #{rx_title}") if datastore['SHOW_TITLES'] == true
```

Comment:
Rex makes the output look nicer, but if a column contains too many characters it overflows to the first column. So if you wanna do that you may need to consider limiting the title size.

Here's an example to the problem (run it in irb and see what happens):

```ruby
      table = Rex::Ui::Text::Table.new(
        'Header'  => 'Test',
        'Indent'  => 1,
        'Columns' => ["Blah", 'Blah', "Blah"]
      )

      table << ['A', 'B' * 1024, 'C']
      table << ['A', 'B', 'C']
      puts table.to_s
```